src/models/subject.ts

```diff
@@ -0,0 +1,157 @@
+import { Classification } from "./decisions/classification";
+import { IDecision, VerificationDecision } from "./decisions/decision";
+import { Verification } from "./decisions/verification";
+
+export type SubjectData = Record<string, unknown>;
+
+export interface ISubjectWrapper {
+    subject: Subject;
+    identifier: symbol;
+    clientCached: boolean;
+    serverCached: boolean;
+
+    url: string;
+    decisions: Map<symbol, IDecision[]>;
+    tag: string;
+}
+
+
+export interface Tag {
+    id?: number;
+    text: string;
+}
+
+export class Subject {
+    public constructor(data: SubjectData) {
+        this.data = data;
+        this.identifier = Symbol("verification-subject");
+    }
+
+    public identifier: symbol;
+    public data: SubjectData;
+}
+
+// the Verification model is emitted by the oe-verification-grid as a CustomEvent
+export class SubjectWrapper implements ISubjectWrapper {
+    public constructor(data: ISubjectWrapper) {
+        this.subject = data.subject;
+        this.decisions = data.decisions ?? new Map();
+        this.url = data.url;
+        this.tag = data.tag;
+
+        this.identifier = Symbol("decision");
+    }
+
+    // aka: context
+    // this is the native data model used by the host application
+    // or this could be the csv row
+    public identifier: symbol;
+    public subject: Subject;
+
+    public clientCached = false;
+    public serverCached = false;
+
+    // each row can only have one verification decision
+    // but can have multiple classification decisions
+    // verification decisions will be reflected in the oe-confirmed
+    // column, while each classification will get its own row
+    public decisions: Map<symbol, IDecision[]>;
+    public url: string;
+    public tag: string;
+
+    public get decisionModels(): IDecision[] {
+        return Array.from(this.decisions.values()).flat();
+    }
+
+    public get verification(): Verification | undefined {
+        return this.decisionModels.find((decision) => decision.type === "verification") as Verification;
+    }
+
+    public get classifications(): Classification[] {
+        return this.decisionModels.filter((decision) => decision.type === "classification") as Classification[];
+    }
+
+    public addDecision(newDecision: IDecision): void {
+        const decision = newDecision;
+
+        // if there are two decisions that are mutually exclusive, we want to
+        // remove them before applying the new decision
+        if (decision instanceof Verification) {
+            const existingVerification = this.verification;
+            if (existingVerification) {
+                this.removeDecision(existingVerification);
+            }
```

Comment:
none of this should be necessary.

```
this.decisions.set(validationKey, decision)
```

would overwrite the value. so why remove it first?